### ISSUE TITLE
Question about gradient computation in Keras related to "Deep Supervision". 

### ISSUE THREAD

**Issue by y-he2** [keras-team/keras#14435]
Hi, 

Firstly I want to give kudos to all you guys who contributed to Keras, amazing works!

Basically "deep supervision" is to set one of the neuron output in the hidden layer as output, and fitting the model with samples containing label for that output. 
I have seem implementation in Keras for works similar to:
https://arxiv.org/abs/1505.02496
https://arxiv.org/pdf/1801.03399.pdf

My question is AFAIK Keras (Tensorflow) is using backpropagation for gradient computation, but that dosent support depending cost function on intermediate outputs?
Is it possible that the gradient computation in Keras is implemented to make it possible to handle "output-label" error calculation for every single neuron?

Best regards, 
Ian

**Comment by sushreebarsa** [comment]
@y-he2 Could you please refer to the [link](https://keras.io/examples/generative/neural_style_transfer/) and let us know if it helps? Thanks!

**Comment by y-he2** [comment]
Hmm thats kinda interesting, it did went into each layers and computed the loss. 
Thanks for example, we can close the issue.
